frontend/src/errorHandling/ErrorContext.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable react/no-multi-comp */
+
+import {
+  Component,
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+} from 'react';
+import { createContext, FC, PropsWithChildren, useState } from 'react';
+import {
+  CustomError,
+  ApolloErrorCatcher,
+  ErrorTypes,
+  SupportedError,
+} from './utils';
+
+interface IErrorContext {
+  errorsQueue: Array<CustomError>;
+  makeErrorCatcher: <T extends SupportedError>(
+    errType: ErrorTypes
+  ) => (err: T) => void;
+  withApolloErrorCatcher: () => ApolloErrorCatcher;
+  getNextError: () => void;
+  execLogin: boolean;
+  setExecLogin: Dispatch<SetStateAction<boolean>>;
+}
+
+interface PropsErrorBoundary {
+  children: ReactNode;
+  makeErrorCatcher: <T extends SupportedError>(
+    errType: ErrorTypes
+  ) => (err: T) => void;
+}
+
+export const ErrorContext = createContext<IErrorContext>({
+  errorsQueue: [],
+  makeErrorCatcher: () => () => null,
+  withApolloErrorCatcher: () => ({ onError: () => () => null }),
+  getNextError: () => null,
+  execLogin: true,
+  setExecLogin: () => null,
+});
+
+interface StateErrorBoundary {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<PropsErrorBoundary, StateErrorBoundary> {
+  private catchError = this.props.makeErrorCatcher(ErrorTypes.RenderError);
+  componentDidCatch(error: Error) {
+    this.catchError(error);
+  }
+  render() {
+    return this.props.children;
+  }
+}
+
+const ErrorContextProvider: FC<PropsWithChildren<{}>> = props => {
+  const { children } = props;
+
+  const [errorsQueue, setErrorsQueue] = useState<Array<CustomError>>([]);
+
+  const [execLogin, setExecLogin] = useState(true);
+  useEffect(() => {
+    if (
+      errorsQueue
+        .map(e => e.getErrorMessage())
+        .map(e => e.includes('Unauthorized'))
+        .includes(true)
```

Comment:
```suggestion
      errorsQueue
        .find(e => e.getErrorMessage().includes('Unauthorized'))
```
Doesn't something alike work?